scripts/docker/compose-up.sh

```diff
@@ -10,9 +22,40 @@ if [ ! -d "certs/" ]; then
     openssl req -x509 -newkey rsa:4096 -keyout certs/key.pem -out certs/cert.pem -sha256 -days 10 -nodes -subj "/C=US/ST=Massachusetts/L=Chelsea/O=PremiScale/CN=platform/emailAddress=$(git config --global --get user.email)" -addext "subjectAltName=DNS:platform"
 fi
 
-# docker compose build -f compose.yaml --no-cache
+
 COMPOSE_DOCKER_CLI_BUILD=1
+export COMPOSE_DOCKER_CLI_BUILD
+
 DOCKER_BUILDKIT=1
-export COMPOSE_DOCKER_CLI_BUILD DOCKER_BUILDKIT
+export DOCKER_BUILDKIT
+
+
+##
+# Reverse lookup the controller name for a given profile.
+function reverse_lookup_profile()
+{
+    if [ $# -ne 1 ]; then
+        printf "ERROR: Function \"reverse_lookup_profile\" expects exactly one argument: Docker compsoe profile name.\\n" >&2
+        exit 1
+    fi
+
+    local controller profile
+
+    profile="$1"
+    mapfile -t controller < <(yq ".services | to_entries[] | select(.value.profiles[0] == \"$profile\") | .key" compose.yaml)
+
+    if [ -z "${controller[*]}" ]; then
+        printf "ERROR: Could not find a controller for the profile \"%s\".\\n" "$1" >&2
+        exit 1
+    elif [ "${#controller[@]}" -ne 1 ]; then
+        printf "ERROR: Found multiple controllers for the profile \"%s\".\\n" "$1" >&2
+        exit 1
+    fi
+
+    printf "%s" "$controller"
+
+    return 0
+}
+
 
-docker compose -f compose.yaml up -d --build premiscale platform echoes registration
\ No newline at end of file
+docker compose --profile "$PROFILE" -f compose.yaml up -d --build "$(reverse_lookup_profile "$PROFILE")" platform echoes registration
```

Comment:
All this work just links the profile together with the name of the service to build.